modules/profile/manifests/jenkins/agent_gpu.pp

```diff
@@ -0,0 +1,101 @@
+# Jenkins Agent Profile
+#
+# Profile class for a node configured to act as a swarm agent for Jenkins.
+# This profile should only ever be declared with an include into a role or site manifest.
+# Parameter overloading should be done using hiera automatic parameter lookup.
+#
+# @example
+#    include profile::jenkins::master
+#
+# @pararm agent_username The unix user the agent will configure and run as.
+class profile::jenkins::agent_gpu {
+
+  include apt
+
+  # neeed for xhost
+  package { 'x11-xserver-utils' :
+    ensure => installed,
+  }
+
+  if $facts['ec2_instance_id'] {
```

Comment:
In order to make use of facts I needed to set `trusted_node_data=true` in the `[main]` stanza of `/etc/puppet/puppet.conf`.

I think this is worth doing and so adding a check to the install_prerequisites.bash script will be necessary.

Reply:
I placed the line in the puppet.conf but while testing changes receiving this error:
```
(err): Evaluation Error: Error while evaluating a Function Call, Generators must be fully qualified at /root/buildfarm_deployment/modules/profile/manifests/jenkins/agent_gpu.pp:41:12 on node ip-10-190-155-70.us-west-1.compute.internal
```

Reply:
I haven't seen that, although I haven't yet refreshed my instance to test on a clean image.

The error is being raised in regards to [line 41](https://github.com/ros-infrastructure/buildfarm_deployment/pull/211/files#diff-dcc43ad42ca303ab8043ec9ad2fbd356R41) which is the start of a later resource and doesn't match the dependency cycle related errors I've seen before (the most likely cause of error when using before and require together).

Reply:
ouch, thanks for the tip. I thought it was the same problem that the first run found about the amazon "facter".  Working now on transforming the generate line (it can not contain special chars so the bash one liners can not work from there) into a custom facter.

Reply:
Should be ready after 18e06402b4ce2e9bf3dab43619646363e2885ce0

Reply:
Opened https://github.com/ros-infrastructure/buildfarm_deployment_config/pull/49 to automatically set the required config directive if it's not yet set.